cmd/create/cluster/cmd.go

```diff
@@ -1145,6 +1172,74 @@ func run(cmd *cobra.Command, _ []string) {
 			}
 		}
 	}
+	oidcEndpointUrl := args.oidcEndpointUrl
+	oidcPrivateKeySecretArn := args.oidcPrivateKeySecretArn
+	isByoOidcSet := false
+	if isSTS {
+		// Support byo oidc config
+		if oidcEndpointUrl != "" && interactive.Enabled() {
+			oidcEndpointUrl, err = interactive.GetString(
+				interactive.Input{
+					Question:   "OIDC Endpoint URL",
+					Help:       cmd.Flags().Lookup(OidcEndpointUrlFlag).Usage,
+					Required:   false,
+					Default:    oidcEndpointUrl,
+					Validators: []interactive.Validator{interactive.IsURL},
+				})
+			if err != nil {
+				r.Reporter.Errorf("Expected a valid OIDC Endpoint Url: %s", err)
+				os.Exit(1)
+			}
+			if oidcEndpointUrl != "" {
+				oidcPrivateKeySecretArn, err = interactive.GetString(
+					interactive.Input{
+						Question: "OIDC Private Key Secret ARN",
+						Help:     cmd.Flags().Lookup(OidcPrivateKeySecretArnFlag).Usage,
+						Required: true,
+						Default:  oidcPrivateKeySecretArn,
+					})
+				if err != nil {
+					r.Reporter.Errorf("Expected a valid ARN to the secret containing the private key: %s", err)
+					os.Exit(1)
+				}
+			}
+		}
+		isByoOidcSet = oidcEndpointUrl != "" || oidcPrivateKeySecretArn != ""
+		if isByoOidcSet {
+			attributesForByoOidc := []string{}
+			if oidcEndpointUrl == "" {
+				attributesForByoOidc = append(attributesForByoOidc, OidcEndpointUrlFlag)
+			}
+			if oidcPrivateKeySecretArn == "" {
+				attributesForByoOidc = append(attributesForByoOidc, OidcPrivateKeySecretArnFlag)
+			}
+			if len(attributesForByoOidc) > 0 {
+				r.Reporter.Errorf("Missing attributes for byo oidc '%s'", helper.SliceToSortedString(attributesForByoOidc))
+				os.Exit(1)
+			}
+			err = interactive.IsURL(oidcEndpointUrl)
+			if err != nil {
+				r.Reporter.Errorf("%s", err)
+				os.Exit(1)
+			}
+			parsedURI, _ := url.ParseRequestURI(oidcEndpointUrl)
+			err := interactive.IsURLReachable(fmt.Sprintf("%s:%s", parsedURI.Host, parsedURI.Scheme))
```

Comment:
I don't think this function belongs in the `interactive` package, as we're not actually using it there. I would just put it in a random helper function elsewhere.